httpcore/src/main/java/org/apache/http/ssl/SSLContextBuilder.java

```diff
@@ -272,6 +274,20 @@ public SSLContext build() throws NoSuchAlgorithmException, KeyManagementExceptio
         return sslcontext;
     }
 
+    public SSLContext buildWithProvider(final Provider provider) throws NoSuchAlgorithmException, KeyManagementException {
```

Comment:
buildWithProvider(final String provider) will throw a new NoSuchProviderException.

in order to be compatible with the old interface so add a new interface